### DISCUSSION TITLE
Using Non-reserved OIDs

### DISCUSSION THREAD

**Discussion by Raytonne** [open-quantum-safe/liboqs#1343] [Ideas]
Currently, except dilithium algorithms have 1.3.6.1.4.1.2.267.x, all other algorithms are using 1.3.9999.x reserved OIDs. Since we manage 1.3.6.1.4.1.54983, we decided to use `1.3.6.1.4.1.54983.2` for QSC algorithms. We simply replace 1.3.9999 with 1.3.6.1.4.1.54983.2; for instance, p256_dilithium2 would be `1.3.6.1.4.1.54983.2.2.7.1`.

Question: Do we want to use non-reserved OIDs? If yes, the OQS Team can manage child OIDs of 1.3.6.1.4.1.54983.2.

Please also check [Issue 409](https://github.com/open-quantum-safe/openssl/issues/409)

**Comment by baentsch** [comment]
> Currently, except dilithium algorithms have 1.3.6.1.4.1.2.267.x, all other algorithms are using 1.3.9999.x reserved OIDs. Since we manage 1.3.6.1.4.1.54983, we decided to use `1.3.6.1.4.1.54983.2` for QSC algorithms. We simply replace 1.3.9999 with 1.3.6.1.4.1.54983.2; for instance, p256_dilithium2 would be `1.3.6.1.4.1.54983.2.2.7.1`.
> 
> Question: Do we want to use non-reserved OIDs? If yes, the OQS Team can manage child OIDs of 1.3.6.1.4.1.54983.2.

Thanks for your offer to move the OQS algorithm OIDs (except dilithium within the IBM managed OID space) into your managed "1.3.6.1.4.1.54983.2" space. Personally I don't see an urgent need for this -- but others may disagree (@dstebila @xvzcf @christianpaquin ?)

**Reply by xvzcf** [reply]
I don't see the need either.

**Comment by dstebila** [comment]
Eventually these will all be assigned OIDs by authoritative owners (either the algorithm creators or NIST), so I'm inclined to wait until those assignments.

**Comment by Raytonne** [comment]
Sounds good; let's wait for the official OIDs then.
